src/codec/rle.rs

```diff
@@ -421,3 +457,46 @@ pub fn rgb565torgb32(input: &[u16], width: usize, height: usize) -> Vec<u8> {
     }
     result_32_bpp
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Cursor;
+
+    use super::process_plane;
+
+    #[test]
+    fn decode_sequence() {
+        // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpegdi/46a9972c-0cdd-4673-add4-87f89b837742
+        let encoded_plane: Vec<u8> = vec![
+            0x13, 0xFF, 0x20, 0xFE, 0xFD, 0x60, 0x01, 0x7D, 0xF5, 0xC2, 0x9A, 0x38, 0x60, 0x01,
+            0x67, 0x8B, 0xA3, 0x78, 0xAF,
+        ];
+        let (width, height) = (6u32, 3u32);
+
+        let mut input_cursor = Cursor::new(encoded_plane);
+
+        // width * height * 4 bytes/pixel
+        let mut output = vec![0 as u8; width as usize * height as usize * 4];
+        process_plane(&mut input_cursor, width, height, &mut output).expect("decode failed");
+
+        // the decoded plane from the MSFT example
+        // (in the doc the scan lines are listed in reverse order)
+        let decoded: Vec<u8> = vec![
+            253, 140, 62, 14, 135, 193, //
+            254, 192, 132, 96, 75, 25, //
+            255, 255, 255, 255, 254, 253, //
```

Comment:
```suggestion
            253, 140, 62, 14, 135, 193,
            254, 192, 132, 96, 75, 25,
            255, 255, 255, 255, 254, 253,
```

Reply:
Without these, rust-analyzer wants to reformat them and put them on 2 lines instead of 3.

The comments are intentional so that the format here matches the example in the spec.

Reply:
Sorry! Just ignore it then.